src/clue.js

```diff
@@ -1,14 +1,124 @@
+        lastName: 'Plum',
+        occupation: 'Designer',
+        age: 22,
+        description: 'Billionaire video game designer',
+        image: 'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg',
+        color: 'purple' 
+    },
+    {   firstName: 'Kasandra',
+        lastName: 'Scarlet',
+        occupation: 'Actor',
+        age: 31,
+        description: 'She is an A - list movie star with a dark past', 
+        image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+        color: 'red'
+    },
+    {   firstName: 'Eleanor',
+        lastName: 'Peacock',
+        occupation: 'Socialité',
+        age: 36,
+        description: 'She is from a wealthy family and uses her status and money to earn popularity',
+        image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+        color: 'blue' 
+    },
+    {   firstName: 'Jack',
+        lastName: 'Mustard',
+        occupation: 'Retired Football player',
+        age: 62,
+        description: 'He is a former football player who tries to get by on his former glory', 
+        image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+        color: 'yellow' 
+    }
+];
 
 // Rooms Collection
-const roomsArray = [];
+const roomsArray = ['Conservatory', 'Dining Room', 'Kitchen', 'Study', 'Library', 
+    'Billiard Room', 'Lounge', 'Ballroom', 'Hall', 'Spa', 'Living Room', 'Observatory', 'Theater', 'Guest House', 'Patio' ];
 
 // Weapons Collection
-const weaponsArray = [];
+const weaponsArray = [
+    { name: 'rope', 
+        weight: 10 
+    },
+    {name: 'knife', 
+        weight: 8
+    },
+    { name: 'candlestick', 
+        weight: 2 
+    },
+    { name: 'dumbbell', 
+        weight: 30
+    },
+    {name: 'poison', 
+        weight: 2 
+    },
+    {name: 'axe', 
+        weight: 15 
+    },
+    {name: 'bat', 
+        weight: 13 
+    },
+    {name:  
+        'trophy', 
+        weight: 25
+    },
+    {name: 'pistol', 
+        weight: 20 
+    }
+];
 
-// ITERATION 2
+function selectRandom(chosenStackArray) {
+    
+    if(chosenStackArray.length === 1) {
+        return chosenStackArray[0]
+    }
+    const randomCard = chosenStackArray[Math.floor(Math.random() * chosenStackArray.length)]
+    return randomCard
+}
+// ITERATION 2 suspectsArray, roomsArray, weaponsArray
+function pickMystery() {
+
+    const selectedSuspect = selectRandom(suspectsArray)
+    const selectedRoom = selectRandom(roomsArray)
+    const selectedWeapon = selectRandom(weaponsArray)
+
+    let selectedCrime = {
+        suspect: selectedSuspect,
+        weapon: selectedWeapon,
+        room: selectedRoom
+    }
+
+    return selectedCrime 
+}
 
 // ITERATION 3
+
+function revealMystery(envelope) {
+
+    return `${envelope.suspect.firstName} ${envelope.suspect.lastName} killed Mr. Boddy using the 
+    ${envelope.weapon.name} in the ${envelope.room}!`
+}
```

Comment:
```suggestion
function revealMystery(envelope) {

    return `${envelope.suspect.firstName} ${envelope.suspect.lastName} killed Mr. Boddy using the ${envelope.weapon.name} in the ${envelope.room.name}!`
}
```
Eran dos cositas muy pequeñas. A envelope.room le faltaba el  `.name` y tenías unos espacios después de "Mr. Boddy using the..."